File: kmp-native/shared/src/commonMain/kotlin/com/stackbenchmark/kmpnative/domain/usecase/GetMovieDetailUseCase.kt
package com.stackbenchmark.kmpnative.domain.usecase

import com.stackbenchmark.kmpnative.domain.entity.MovieDetail
import com.stackbenchmark.kmpnative.domain.repository.MovieRepository

class GetMovieDetailUseCase(private val repository: MovieRepository) {
    suspend operator fun invoke(id: Int): Result<MovieDetail> =
        repository.getMovieDetail(id)
}

File: rn-expo/src/application/GetMovieDetailUseCase.ts
import { MovieDetail } from "../domain/entities";
import { MovieRepository, Result } from "../domain/MovieRepository";

export class GetMovieDetailUseCase {
  constructor(private readonly repository: MovieRepository) {}

  async execute(id: number): Promise<Result<MovieDetail>> {
    return this.repository.getMovieDetail(id);
  }
}
